tests/unit/lib/test_opensearch_secrets.py

```diff
@@ -0,0 +1,169 @@
+        self.secret_store.put_object(
+            scope, "key-obj", {"name1": None, "name2": "val2"}, merge=True
+        )
+        self.assertDictEqual(self.secret_store.get_object(scope, "key-obj"), {"name2": "val2"})
+
+    @override
+    @parameterized.expand([(Scope.APP), (Scope.UNIT)])
+    def test_nullify_obj(self, scope):
+        """Test iteratively filling up an object with `None` values."""
+        self.secret_store.put_object(scope, "key-obj", {"key1": "val1", "key2": "val2"})
+        self.secret_store.put_object(scope, "key-obj", {"key1": None, "key2": "val2"}, merge=True)
+        self.secret_store.put_object(scope, "key-obj", {"key2": None}, merge=True)
+        self.assertFalse(self.secret_store.has(scope, "key-obj"))
+
+    def test_label_app(self):
+        scope = Scope.APP
+        label = self.secret_store.label(scope, "key1")
+        self.assertEqual(label, f"opensearch:{scope}:key1")
+        self.assertEqual(
+            self.secret_store.breakdown_label(label),
+            {"application_name": "opensearch", "scope": scope, "unit_id": None, "key": "key1"},
+        )
+
+    def test_label_unit(self):
+        scope = Scope.UNIT
+        label = self.secret_store.label(scope, "key1")
+        self.assertEqual(self.secret_store.label(scope, "key1"), f"opensearch:{scope}:0:key1")
+        self.assertEqual(
+            self.secret_store.breakdown_label(label),
+            {"application_name": "opensearch", "scope": scope, "unit_id": 0, "key": "key1"},
+        )
+
+    @parameterized.expand([(Scope.APP), (Scope.UNIT)])
+    def test_save_secret_id(self, scope):
+        """Test putting and getting objects in/from the secret store."""
+        self.secret_store.put(scope, "key", "val1")
+        secret_id = self.secret_store._get_relation_data(scope)[
+            self.secret_store.label(scope, "key")
+        ]
+        secret_content = self.charm.model.get_secret(id=secret_id).get_content()
+        self.assertEqual(secret_content["key"], "val1")
+
+        self.secret_store.put_object(scope, "key-obj", {"name1": "val1"}, merge=True)
+        secret_id2 = self.secret_store._get_relation_data(scope)[
+            self.secret_store.label(scope, "key-obj")
+        ]
+        secret_content = self.charm.model.get_secret(id=secret_id2).get_content()
+        self.assertEqual(secret_content["name1"], "val1")
+
+    def test_bad_label(self):
+        with self.assertRaises(IndexError):
+            self.secret_store.breakdown_label("bla")
+
+        with self.assertRaises(IndexError):
+            self.secret_store.breakdown_label("bla-bla-bla")
+
+        with self.assertRaises(UnboundLocalError):
+            self.secret_store.breakdown_label("bla:bla")
+
+        with self.assertRaises(UnboundLocalError):
+            self.secret_store.breakdown_label("bla:bla:bla")
+
+        with self.assertRaises(ValueError):
+            self.secret_store.breakdown_label("bla:bla:bla:bla")
```

Comment:
Hmmm, I thought I added this question already (if duplicate: sry).

@Mehdi-Bendriss this is the test I added as we agreed on our live chat.
Are we 100% sure that this is what we want to have?

Reply:
You can keep this test - as discussed [here](https://github.com/canonical/opensearch-operator/pull/102#discussion_r1302800610). 
TL;DR: we *should not* expect an arbitrary input to this method, because the label creation / formatting is handled by the `label()` method. So if something like this happens we definitely have a problem elsewhere. For debugging purposes we can keep it - IF juju secrets_changed event's data are somewhat invalid..